addons/website_sale/models/sale_order.py

```diff
@@ -353,13 +349,49 @@ def action_recovery_email_send(self):
                 'default_composition_mode': 'mass_mail' if len(self.ids) > 1 else 'comment',
                 'default_res_id': self.ids[0],
                 'default_model': 'sale.order',
-                'default_use_template': bool(template_id),
-                'default_template_id': template_id,
+                'default_use_template': bool(template),
+                'default_template_id': template.id if template else False,
                 'website_sale_send_recovery_email': True,
                 'active_ids': self.ids,
             },
         }
 
+    @api.multi
+    def _get_recovery_template(self):
+        """
+        Return the cart recovery template for a set of orders.
+        If they all belong to the same website, we return the default template;
+        otherwise we return the website-specific template.
+        """
+        try:
+            default_template = (self.env.ref('website_sale.mail_template_sale_cart_recovery', raise_if_not_found=False)
+                                or self.env['mail.template'])
+            template = (self.filtered('website_id') == self and
+                        self.mapped('website_id')[-1:1].cart_recovery_mail_template_id) or default_template
+        except:
+            template = self.env['mail.template']
+        return template
+
+    @api.multi
+    def mark_cart_recovery_email_sent(self):
```

Comment:
Do we need this method to be public?

Also missing docstring.

And this method is strange, why/when would we want to call it on records that we need to filter because we don't want to mark cart recovery email sent?